web/components/PetPostModal/PetPostModal.tsx

```diff
@@ -9,12 +9,93 @@ import {
   Text,
 } from "@chakra-ui/react";
 import ImageSlider from "./ImageSlider";
-import PetPostTagGroup from "./PetPostTagGroup";
+import PetPostListGroup from "./PetPostListGroup";
+import {
+  fosterTypeLabels,
+  behavioralLabels,
+  spayNeuterStatusLabels,
+  IPost,
+  medicalLabels,
+  crateTrainedLabels,
+  houseTrainedLabels,
+  GoodWith,
+  Trained,
+  goodWithLabels,
+  genderLabels,
+  breedLabels,
+  sizeLabels,
+  ageLabels,
+  fosterTypeDescriptions,
+} from "../../utils/types/post";
 
 const PetPostModal: React.FC<{
   isOpen: boolean;
   onClose: () => void;
-}> = ({ isOpen, onClose }) => {
+  postData: IPost;
+}> = ({ isOpen, onClose, postData }) => {
+  const {
+    name,
+    description,
+    type,
+    size,
+    age,
+    spayNeuterStatus,
+    houseTrained,
+    crateTrained,
+    behavioral,
+    medical,
+    gender,
+    breed,
+    getsAlongWithOlderKids,
+    getsAlongWithYoungKids,
+    getsAlongWithLargeDogs,
+    getsAlongWithSmallDogs,
+    getsAlongWithWomen,
+    getsAlongWithMen,
+    getsAlongWithCats,
+    attachments,
+  } = postData;
+
+  const behavioralTagLabels: string[] = behavioral?.map(
+    (tag) => behavioralLabels[tag]
+  );
+
+  const medicalTagLabels: string[] = medical.map((tag) => medicalLabels[tag]);
+
+  const trainedTagLabels: string[] = [
+    spayNeuterStatusLabels[spayNeuterStatus],
+    houseTrainedLabels[houseTrained],
+    crateTrainedLabels[crateTrained],
+  ];
+
+  const behavioralAndMedicalTagLabels =
+    medicalTagLabels.concat(behavioralTagLabels);
+
+  const behavioralMedicalAndTrainedTagLabels =
+    behavioralAndMedicalTagLabels.concat(trainedTagLabels);
+
+  const goodWithLabelsAndValues = [
+    [goodWithLabels[GoodWith.Men], getsAlongWithMen],
+    [goodWithLabels[GoodWith.Women], getsAlongWithWomen],
+    [goodWithLabels[GoodWith.OlderChildren], getsAlongWithOlderKids],
+    [goodWithLabels[GoodWith.YoungChildren], getsAlongWithYoungKids],
+    [goodWithLabels[GoodWith.LargeDogs], getsAlongWithLargeDogs],
+    [goodWithLabels[GoodWith.SmallDogs], getsAlongWithSmallDogs],
+    [goodWithLabels[GoodWith.Cats], getsAlongWithCats],
+  ];
+
+  const goodWithTagLabels: string[] | undefined = goodWithLabelsAndValues
+    .filter((goodWith) => goodWith[1] === Trained.Yes)
+    .map((goodWith) => goodWith[0]);
+
+  const notGoodWithTagLabels: string[] | undefined = goodWithLabelsAndValues
+    .filter((goodWith) => goodWith[1] === Trained.No)
+    .map((goodWith) => goodWith[0]);
+
+  const unsureAboutTagLabels: string[] | undefined = goodWithLabelsAndValues
+    .filter((goodWith) => goodWith[1] === Trained.Unknown)
+    .map((goodWith) => goodWith[0]);
```

Comment:
For readability, make two maps instead of the nested array structure:

1 map defines which `getsAlongWith...` field maps to each enum that is passed in as props:
```js
const goodWithLabelMap = {
  getsAlongWithMen: GoodWith.Men,
  getsAlongWithWomen: GoodWith.Women,
  ...
}
```

The second map defines how each `getsAlongWith...` field maps to each passed in value:
```js
const goodWithValueMap = {
  getsAlongWithMen,
  getsAlongWithWomen,
  ...
}
```
Note that this syntax automatically makes the key of each element the same name as the variable, i.e. `"getsAlongWithMen": Trained.Yes` is a possible field.

Then, make an array of all `getsAlongWith...` fields like
```js
const getsAlongWithFields = ["getsAlongWithMen", "getsAlongWithWomen", ... ];
```
Do something like this for good with, not good with, and unsure:
```js
const goodWithTagLabels = getsAlongWithFields.filter((field) => goodWithValueMap[field] === Trained.Yes).map((field) => goodWithLabelMap[field])
...
```

This may seem a bit verbose and excessive, but I think it will be helpful down the line!